### DISCUSSION TITLE
PHP to Python Exodus

### DISCUSSION THREAD

**Discussion by davidjeba** [django-cms/django-cms#7166] [General]
5Years back in India, almost every Computer Science Student who has graduated came out with the knowledge of PHP and Wordpress
I happened to interview like 200 students last month, almost 95% practiced Python [unfortunately without CMS framework knowledge]

Tech scape is shifting, PHP developers are also finding python to be easy and interesting but anyone who has got into wordpress cannot come out because theres no equivalent CMS for Wordpress.

Can django cms 4.0 mimic wordpress for a while?
django editor is a nightmare, it would be easy for everyone to just shamelessly follow wordpress like admin architecture
posts, pages, plugins, themes, setting, gutentor.com etc

Target PHP developers who has tons of themes and plugins in themeforest like portals to take a little effort to rewrite their themes and plugins for django-press that would exponentially increase the adaption of django
integrating svelte, tailwind and cloudflare natively in django 4.0 with native PWA support would be super huge advantages for the ecosystem

**Comment by jrief** [comment]
What do you mean by

> django editor is a nightmare, it would be easy for everyone to just shamelessly follow wordpress like admin architecture

Neither Django, nor Django-CMS provide an editor. djangocms-text-ckeditor does, and it's just a rich-text editor written in JavaScript. If in your opinion that editor is a nightmare, then that's a problem of that editor, not Django's or its CMS. I agree that [CKEditor](https://ckeditor.com/) has some flaws, but it's still very powerful and I haven't any way better alternative yet.

> exponentially increase the adaption of django
integrating svelte, tailwind and cloudflare natively in django 4.0 with native PWA support would be super huge advantages for the ecosystem

Svelte is a compiler to create web-components. You can perfectly integrate that with Django.

Tailwind is a CSS framework plus kind-of compiler. You also can perfectly integrate that with Django.

If you need a PWA, you presumably need websockets. [Django-Channels](https://channels.readthedocs.io/en/stable/) provides a very good websocket support. So what exactly is your problem with PWAs?

**Reply by davidjeba** [reply]
Imagine, im not a developer. I still can launch a website on wordpress, install beautiful themes, extend with plugins 
but with Django its not easy. In wordpress it never mattered to me whether its a Ckeditor or javascript compiler. all i know is Code editor, block editor, frontend editor [elementor] thats it.

my question is, if Django is focussed towards non-developer as a low-code platform 
more people will try it and use it.
i understand its a framework. but Wix, squarepages, wordpress.com are all gaining momentum.
django should have a version of itself targeting school students who can launch a website without knowing a thing.

Django is 1000% powerful than Wordpres, i Agree!
but Wordpress is Million % EASIER than rest of the world
thats my point...

FYI, I train students, freshers on basics of website building. and this is my feedback

**Comment by NicolaiRidani** [comment]
Hi @davidjeba, thanks for opening this discussion.  This is indeed a recurring discussion in the community and I agree with your observation regarding the trend of PHP to Python (see this [graph](https://insights.stackoverflow.com/trends?tags=php%2Cpython)). But we at the django CMS project (important: we represent django CMS, not Django) have a clear stance on this (at least for the time being). And to make my life easier, I'll quote Daniele, who summed it up perfectly in the comments section of an article a few years ago:

> We're very conscious of that advantage of WordPress/PHP/Joomla - we mention it right at the top of the article!
But with django CMS and Aldryn we're not aiming to serve the same market. django CMS is an enterprise-level product, and can't ever be as simple to get started with as (say) Joomla or WordPress.
Organisations and developers that adopt django CMS need things that WordPress can't provide - so our aim is to make life easier for them, rather than to try to meet the needs of every WordPress user in the world

BUT...of course, that doesn't mean we don't need to constantly improve django CMS and make life easier for developers and editors alike...it's just important to understand where we position django CMS compared to other tools.

**Reply by davidjeba** [reply]
Nicolai,
I agree with your point of view and mission of Django cms.
QTM - Quick to Market is one key thing that is applicable for Enterprise too.
I have myself used, proposed Django framework for enterprise as they can build anything on it
but Django CMS is a CMS which is built to manage content, websites, etc not an ERP or any other application
DjangoCMS by itself doesn't have to provide everything, something like Elementor for WordPress kind of integration would make it better as two companies who can complement each other to make it better
and How woocommerce complements WordPress something like that can make django cms better.
I'm not against your directions, vision, and mission
I love Django a lot but could not use/promote much of it
